### ISSUE TITLE
Bug when having lists with and without key-values

### ISSUE THREAD

**Issue by justanotherjurastudent** [christianwannerstedt/obsidian-key-value-list#24]
Hi,
I noticed following problem with this list example:

- first: First
- second: Second
- some random text

The first two list points are key values, the last is just normal text. The key-values are just formatted rightly when positioning the cursor on the last bullet point. When I leave the list, there is no correct formatting for the key values. There is even no formatting in read-mode.

I've made this video in my existing project. I set the delimiter for key values as a "_".

https://github.com/user-attachments/assets/d1aac3c6-584c-40bd-a800-9d0fa6e1aa41

I think, this behaviour is beacuse of this code line:
```
if (!this.isKeyValueListItem(line2) && cursorLine !== listEndLineLookup) {
            isKeyValueList = false;
          }
```
It could be, that there is a bug with _cursorLine !== listEndLineLookup_
I want that the key values render correctly if there is a further bullet point after or before that.
I've tried to fix that but I don't get it right. Can you help?

**Comment by harrislop** [comment]
I have a question that’s unrelated to plugins. How did you apply the indent highlight you're using? I'm curious if it's done with CSS or a plugin. It looks really amazing!

**Comment by justanotherjurastudent** [comment]
> I have a question that’s unrelated to plugins. How did you apply the indent highlight you're using? I'm curious if it's done with CSS or a plugin. It looks really amazing!

Yeah, i use the bullet threading css-snippet (https://gist.github.com/KillyMXI/cbef8edff6dd55d9e6ea4df66567e9b1). You can add this css-files into obsidian. I prefer my own version of this code, but first try his code ;)

**Comment by harrislop** [comment]
> > I have a question that’s unrelated to plugins. How did you apply the indent highlight you're using? I'm curious if it's done with CSS or a plugin. It looks really amazing!
> 
> Yeah, i use the bullet threading css-snippet (https://gist.github.com/KillyMXI/cbef8edff6dd55d9e6ea4df66567e9b1). You can add this css-files into obsidian. I prefer my own version of this code, but first try his code ;)

Thank you! I wish you many wonderful days ahead!

**Comment by christianwannerstedt** [comment]
Hi! Sorry for the late follow-up. I’ve decided to dust off this plugin and am currently working my way through all the open issues.

What you’re describing isn’t actually a bug, but an intentional design decision. As stated in the README:

“In order for a list to be considered a Key-Value list by this plugin, EVERY row must contain a key-value pair. If one or more rows don’t contain a key-value pair, the list will be ignored by the plugin.”

The reason for this is to avoid reformatting individual lines just because they happen to contain the delimiter character. Requiring every row to be a valid key-value pair makes it much less likely that the plugin will accidentally modify content that wasn’t intended to be treated as a key-value list.
